elyra/pipeline/component.py

```diff
@@ -114,21 +114,27 @@ class Component(object):
     Represents a generic or runtime-specific component
     """
 
-    def __init__(self, id: str, name: str, description: Optional[str], source_type: str,
-                 source: str, catalog_entry_id: str, runtime: Optional[str] = None, op: Optional[str] = None,
-                 category_id: Optional[str] = None, properties: Optional[List[ComponentParameter]] = None,
+    def __init__(self, id: str, name: str,
+                 description: Optional[str],
+                 source_type: str,
+                 source: str,
```

Comment:
I think we should try to match schema (for schema match code) so would recommend `location_type` and `location` for these.